main.go

```diff
@@ -238,6 +247,34 @@ func initClient() {
 	log.Debug(string(output))
 }
 
+func startReaperWorker() {
+	// See related issue in go-reaper https://github.com/ramr/go-reaper/issues/11
+	if _, hasReaper := os.LookupEnv("REAPER"); !hasReaper {
+		go reaper.Reap()
+
+		args := append(os.Args, "#worker")
+
+		pwd, err := os.Getwd()
+		if err != nil {
+			panic(err)
+		}
+
+		workerEnv := []string{fmt.Sprintf("REAPER=%d", os.Getpid())}
+
+		var wstatus syscall.WaitStatus
+		pattrs := &syscall.ProcAttr{
+			Dir:   pwd,
+			Env:   append(os.Environ(), workerEnv...),
+			Sys:   &syscall.SysProcAttr{Setsid: true},
+			Files: []uintptr{0, 1, 2},
+		}
+		workerPid, _ := syscall.ForkExec(args[0], args, pattrs)
+
+		syscall.Wait4(workerPid, &wstatus, 0, nil)
+		return
```

Comment:
Just a small nit, since you put this code into a function, you don't want to return back to `main` and continuing running the rest of the `main` code in the parent process. 

The child process will run it because it will not get into this `if` code block but the parent really doesn't need to (or maybe even shouldn't). 
Either return a boolean back to `main` (true/false to continue executing the rest of the main code or return out of main in the parent case) or you can then just call `os.Exit(0)` here in lieu of a return.

Reply:
I missed that one. Maybe that's the source of the odd behavior I was getting. I'll fix that now. Thanks for review :)

Reply:
For some reason, it misbehaves if I exit at this point. I'll give it a bit more of debugging.

Reply:
I wonder if it is getting an error code indicated an interrupted system call. You could print the error from `wait4` and check.
And if that's the case, replace that wait call with something like: 
```
_, err := syscall.Wait4(workerPid, &wstatus, 0, nil)
for syscall.EINTR == err {
		_, err = syscall.Wait4(workerPid, &wstatus, 0, nil)
}
```

HTH

Reply:
When I print the error coming from syscall.Wait4 I also get `no child processes`

Reply:
That error `no child processes` is actually ok. Because there is also the possibility the reaper can complete the `wait` on the child process instead of this call. In that case you'd get that error.

It is only in the case of `EINTR` that you would need to retry the system call.

Reply:
Even when it's not the case, would yo recommend to add it?

Reply:
Yeah, I'd still recommend adding that `for loop wait` in there on `EINTR` as it is still possible to reach that part of the code and you don't want to terminate the process early.

Reply:
What's the bug you are hitting or the symptom of the problem you have?

Reply:
I'm having no other problems. I still get a `no child process` when waiting for the `lizardfs` process but the process actually finishes correctly and the tests are passing so I'm just logging it for reference. Technically everything is working fine. I'll add the loop to catch this problem and create another PR.
Thank you so much for your help :)